File: GoFrServer/docs/docs.py
import os

# Directory containing the files to concatenate
directory = '/Users/yuktha/Downloads/gofr-development/docs'
output_file = 'output.txt'

# Open the output file in write mode
with open(output_file, 'w') as outfile:
    # Iterate over each file in the directory
    for filename in os.listdir(directory):
        # Get the full file path
        file_path = os.path.join(directory, filename)
        
        # Only process files (not directories)
        if os.path.isfile(file_path):
            with open(file_path, 'r') as infile:
                # Write the contents of the current file to the output file
                outfile.write(infile.read())
                outfile.write("\n")  # Add a newline between file contents

print(f"All files in {directory} have been concatenated into {output_file}")

File: gofr-server/docs/docs.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"path/filepath"
)

func main() {
	// Specify the directory containing .md files
	dir := "/Users/yuktha/Downloads/gofr-development/docs" // Set your directory here

	// Create the output file
	outputFile, err := os.Create("combined.md")
	if err != nil {
		fmt.Println("Error creating output file:", err)
		return
	}
	defer outputFile.Close()

	// Read all files in the directory
	err = filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
		if err != nil {
			return err
		}
		// Check if the file is a .md file
		if filepath.Ext(path) == ".md" {
			// Read the content of the file
			content, err := ioutil.ReadFile(path)
			if err != nil {
				fmt.Println("Error reading file:", err)
				return err
			}

			// Write the content to the output file
			_, err = outputFile.Write(content)
			if err != nil {
				fmt.Println("Error writing to output file:", err)
				return err
			}

			// Add a newline between files
			_, err = outputFile.WriteString("\n")
			if err != nil {
				fmt.Println("Error writing newline:", err)
				return err
			}
		}
		return nil
	})

	if err != nil {
		fmt.Println("Error walking through directory:", err)
		return
	}

	fmt.Println("Files concatenated successfully into 'combined.md'")
}
